src/metric/AlphaPrecisionSampleMetricDescription.py

```diff
@@ -0,0 +1,134 @@
+from src.util.CudaUtil import get_default_device, to_device
+import src.util.ModelUtil as ModelUtil
+
+import torch
+import torchvision.transforms as T
+
+#! Constants
+PARAM_ALPHA_DEFAULT = 1.0
+
+SETUP_MODE_RESTART = "restart"  # Restart setup from beginning
+SETUP_MODE_CONTINUE = "continue"  # Continue training of evaluation embedding
+SETUP_MODE_PROJECT = "project"  # Skip training and project immediately
+
+
+class AlphaPrecisionSampleMetricDescription(SampleMetricDescription):
+    @staticmethod
+    def get_name() -> str:
+        return "AlphaPrecision"
+
+    @staticmethod
+    def setup(dataset: Dataset, mode: str = SETUP_MODE_CONTINUE) -> None:
+
+        # Check if valid mode
+        if mode not in (SETUP_MODE_RESTART, SETUP_MODE_CONTINUE, SETUP_MODE_PROJECT):
+            raise ValueError(f"Invalid setup mode: '{mode}'")
+
+        # Train evaluation embedding if applicable
+        if mode != SETUP_MODE_PROJECT or EE.get() is None:
+
+            # Sanity check for projection mode
+            if mode == SETUP_MODE_PROJECT:
+                print("Must train evaluation embedding before projection!")
+                print("Changing mode to perform setup from scratch.")
+                mode = SETUP_MODE_RESTART
+
+            # Train network
+            info = None
+            if mode == SETUP_MODE_CONTINUE:
+                info = ModelUtil.load_aux_best(EE.AUX_MODEL_NAME)
+                print(f"Starting from (epoch, batch) = ({info.epoch}, {info.batch})")
+            EE.train(dataset, info)
+
+        # Project samples
+        EE.project(dataset)
+
+    @staticmethod
+    def is_ready(dataset: Dataset) -> bool:
+        return (
+            EE.get() is not None
+            and ModelUtil.get_file_jar().get_file(
+                AlphaPrecisionSampleMetricDescription._ds_proj_file_name(dataset),
+                np.load,
+            )
+            is not None
+        )
+
+    @staticmethod
+    def calc(data: pd.DataFrame, dataset: Dataset, **parameters: Any) -> np.ndarray:
+        ee = EE.get()
+
+        # Sanity check
+        if ee is None:
+            raise ValueError(
+                "Evaluation embedding is not available! Make sure"
+                + " that the setup has been performed."
+            )
+
+        # Load device
+        device = get_default_device()
+
+        # Send to device
+        ee = to_device(ee, device)
+
+        # Extract parameters
+        alpha = parameters["alpha"] if "alpha" in parameters else PARAM_ALPHA_DEFAULT
+
+        # Load population images
+        pop_images = TorchImageDataset(
+            [p for p in data.iloc[:, 2]],
+            T.Compose(
+                [
+                    T.ToTensor(),
+                    T.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5)),
+                ]
+            ),
+        )
+
+        # Create population loader
+        pop_loader = torch.utils.data.DataLoader(
+            pop_images, batch_size=1, shuffle=False, num_workers=2
```

Comment:
is batch_size = 1 reasonable? or just that low for testing?

Reply:
This was temporary, I forgot to add support for batch size